web/siteplan/src/util/Controls/ExportControl.ts

```diff
@@ -50,182 +70,417 @@ export default class ExportControl extends Control {
+    tileLayer.getSource()?.addFeatures(featuresInsideSheetCut)
+    return tileLayer
+  }
 
-    if (!size || !viewResolution) {
+  private drawTileLayer (
+    mapLayer: HTMLCanvasElement,
+    canvas: HTMLCanvasElement,
+    x: number,
+    y: number,
+    viewsize: number[]
+  ) {
+    if (mapLayer.width <= 0) {
       return
     }
 
-    this.map.setSize([width, height])
-    const scaling = Math.min(width / size[0], height / size[1])
-    this.map.getView().setResolution(viewResolution / scaling)
+    const context = canvas.getContext('2d')
+    if (context === null) {
+      return
+    }
+
+    context.clearRect(x, y, viewsize[0], viewsize[1])
+    context.fillStyle = 'white'
+    context.fillRect(x, y, viewsize[0], viewsize[1])
+
+    const matrix = this.getMapTransformMatrix(mapLayer)
+
+    // Apply the transform to the export map context
+    context.setTransform(matrix[0], matrix[1], matrix[2], matrix[3], matrix[4], matrix[5])
+
+    const backgroundColor = (mapLayer.parentNode as HTMLHtmlElement | null)?.style.backgroundColor
+    if (backgroundColor) {
+      context.fillStyle = backgroundColor
+      context.fillRect(0, 0, mapLayer.width, mapLayer.height)
+    }
+
+    const xOffset = x * matrix[0] + y * matrix[1]
+    const yOffset = x * matrix[2] + y * matrix[3]
+    context.drawImage(mapLayer, xOffset, yOffset)
+    // Reset transform
+    context.setTransform(1, 0, 0, 1, 0 ,0)
+  }
+
+  private getMapTransformMatrix (mapCanvas: HTMLCanvasElement) {
+    let result = [1, 0, 0, 1, 0, 0]
+    const transform = mapCanvas.style.transform
+    if (transform) {
+      // Get the transform parameters from the style's transform matrix
+      const macht = transform.match(/^matrix\(([^\(]*)\)$/)
+      if (macht && macht[1] !== null)
+        result = macht[1].split(',').map(Number)
+    }
+
+    return result
+  }
+
+  /**
+   * Resize tiles canvas to fit sheet cut polygon
+   * @param tilesCanvas the canvas
+   * @param sheetCutPolygon the sheet cut polygon, which rotation to the nord pole
+   * @param resolution the map resolution
+   */
+  private resizeTilesCanvas (
+    tilesCanvas: HTMLCanvasElement | undefined,
+    sheetCutPolygon: Polygon,
+    resolution: number
+  ): HTMLCanvasElement | undefined {
+    const exportCanvas = document.createElement('canvas')
+    // Currently the polygon is rotation to the nord pole.
+    // To export sheet cut in vertical, should change the width/height of polygon
+    exportCanvas.width = getHeight(sheetCutPolygon.getExtent()) / resolution
+    exportCanvas.height = getWidth(sheetCutPolygon.getExtent()) / resolution
+    const exportCtx = exportCanvas.getContext('2d')
+    if (!exportCtx || !tilesCanvas) {
+      return
+    }
+
+    exportCtx.fillStyle = 'white'
+    exportCtx.fillRect(0,0, exportCanvas.width, exportCanvas.height)
+
+    // Rotate the tile canvas to 90
+    exportCtx.rotate(-Math.PI / 2)
```

Comment:
@TruongQuangSB Why not use a more readable format

```suggestion
    exportCtx.rotate(toRad(-90))
```